crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -793,7 +787,7 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
     ///
     /// If all children are valid, then this essentially moves appends all children blocks to the
     /// new block's chain.
-    fn try_connect_buffered_blocks(&mut self, new_block: BlockNumHash) {
+    fn try_connect_all_buffered_blocks(&mut self, new_block: BlockNumHash) {
```

Comment:
So the idea is to call this function only on `restore_canonical_hashes` when we are doing the switch from pipeline to the tree as this is expensive (takes time). And on an ordinary block call `append_buffered_block_one` that is going to insert only one block on the tip of the chain and doesn't block.

Reply:
> So the idea is to call this function only on restore_canonical_hashes when we are doing the switch from pipeline to the tree as this is expensive (takes time).

I'm not familiar with this code and didn't really understand it.

> on the tip of the chain and doesn't block.
it only blocks for 1 block execution, just as on new payload